ehr/resources/web/ehr/panel/AnimalHistoryPanel.js

```diff
@@ -82,6 +82,58 @@ Ext4.define('EHR.panel.AnimalHistoryPanel', {
         this.createTabPanel();
     },
 
+    displayWarning: function(tab){
+        var status = Ext4.create('Ext.Component',{
+            itemId: 'reportMessage',
+            html: '<div class="alert alert-warning">' + tab.report.warning + '</div>'
+        });
+
+        tab.add(status);
+    },
+
+    displayReport: function(tab){
+        // If we have a status to show the user to help set expectations, display it at the top
+        if (tab.report.reportStatus) {
+            var status = Ext4.create('Ext.Component',{
+                itemId: 'reportStatus',
+                html: '<div class="alert alert-warning" role="alert">Report Status: <strong>'
+                        + Ext4.util.Format.htmlEncode(tab.report.reportStatus)
+                        + '</strong></div>'
+            });
+
+            tab.add(status);
+        }
+
+        if (tab.report.warning) {
```

Comment:
From what I can tell, this if statement is the only difference from TabbedReportPanel.displayReport(). Can we consolidate code? Could we replace the TabbedReportPanel implementation with this one? All reports without a warning should be unaffected, right?

Reply:
As noted on the email to Marty I also have concerns and would favor simply trying to make validation asynchronous.  I sent a proposal on email, but we can shift the conversation here is preferred.